### PR TITLE
Toolbar

### PR DESCRIPTION
**Pull request by pshriwise**
Adds a locked toolbar object to the side of the window. Objects are cloned as they are dragged into the open space. 

Its still a little buggy as objects can get stuck in the toolbar area and become new 'source' objects.

Another hack was making the toolbar object a perfect absorber so it doesn't appear to mess with the physics. If there's a better way to do that, let me know.

### CODE REVIEW

File: main.cc [hunk 1]
```diff
@@ -154,6 +152,13 @@ int main(int argc, char** argv) {
         } else if (ev.type == SDL_MOUSEBUTTONDOWN &&
                    ev.button.button == SDL_BUTTON_LEFT) {
           dragged = sys.ObjectFor(ev.button.x, ev.button.y);
+	  if(dragged->isToolbar())
+	    continue;
```
Comment (rwcarlsen) [inline main.cc:156]: This skips the part of the main loop that does stuff other than process events (i.e. update screen render, move neutrons, etc.) - need to use alternate way to skip without omitting code outside event processing if-elseif.

Comment (rwcarlsen) [inline main.cc:156]: Never mind - mixed up my scopes.

**Comment by rwcarlsen** [comment]
Looks great!  I'd like to have a way to delete squares (e.g. drag and release them onto the toolbar) and also the detectors, streaming source, voidcoeff moderator, etc to all be on the toolbar too before merging.  Were you planning on working on those, or would you prefer to hand it over?

**Comment by pshriwise** [comment]
I just finished adding the trash capability. Should fix the problem of people creating new spawn items in the toolbar as well. I'll push those changes in a sec. 

I'll leave the other material additions & placement on the toolbar to you. I remember having some trouble with detectors, but I can't recall why.

**Comment by rwcarlsen** [comment]
Did you forget to push trash capability commits - or are you still working on them?

**Comment by rwcarlsen** [comment]
I got the detector stuff figured out - missing clone method and also a bug in InToolbar function.

**Pull request merged**